Task1/todo.js

```diff
@@ -0,0 +1,121 @@
+// This is the array that will hold the todo list items
+let todoItems = [];
+// This function will create a new todo object with text input, 
+//check to see which task is completed and id for remove
+
+
+// Render the todo items
+function renderTodo(todo)
+{
+    const list=document.querySelector("#todoList");
+    const item=document.querySelector(`[todo-id='${todo.id}']`);
+
+    if(todo.deleted)
+    {
+        item.remove();
+        return
+    }
+    const node=document.createElement("li");
+    node.setAttribute('todo-id',todo.id);
+    const deleteButton =document.createElement("button");
+    deleteButton.classList.add('js-delete-todo');
+    deleteButton.innerHTML="Delete" 
+    node.innerHTML = `
+    <span>${todo.text}</span>
+    `;
+    node.appendChild(deleteButton)
+    list.appendChild(node)
+}
+function addTodo(text)
+{
+    const todo={
+        text,
+        id:Date.now(),
+    };
+
+    todoItems.push(todo);
+    renderTodo(todo);
+}
+
+function deleteTodo(id)
+{
+    const index=todoItems.findIndex(item=>item.id===Number(id));
+    const todo={
+        deleted:true,
+        ...todoItems[index]
+    };
+    todoItems=todoItems.filter(item=>item.id!==Number(id));
+    renderTodo(todo);
+}
+//Select the form element
+const form = document.querySelector("#todoForm");
+//Add event listener
+const add = document.querySelector("#button");
+add.addEventListener('click',event=>{
+    //prevent page from refresh on clicking
+    event.preventDefault();
+    //select the input
+    const input=document.querySelector("#todoInput");
+    //get value of input and remove whitespaces we can use createNodetext also
+    const text=input.value.trim();
+    if(text!=='')
+    {
+        addTodo(text);
+        //reset value after pushing to ''
+        input.value='';
+        //give focus to element
+        input.focus();
+    }
+});
+
+const list=document.querySelector('#todoList');
+list.addEventListener('click',event=>{
+    if(event.target.classList.contains('js-delete-todo'))
+    {
+        const itemKey=event.target.parentElement;
+        deleteTodo(itemKey.getAttribute('todo-id'));
+    }
+});
+
+const deleteall=document.querySelector('.delete-all');
+deleteall.addEventListener('click',event=>{
+    todoItems.splice(0);
```

Comment:
Try to use a different approach here to remove/empty items from the array instead of `.splice()`.

@Poorvi-3pillar